config/confighttp/confighttp.go

```diff
@@ -105,10 +105,15 @@ type ClientConfig struct {
 // We encourage to use this function to create an object of ClientConfig.
 func NewDefaultClientConfig() ClientConfig {
 	// The default values are taken from the values of 'DefaultTransport' of 'http' package.
+
+	// Configure with the non-nil map value.
+	headers := make(map[string]configopaque.String)
+
 	maxIdleConns := 100
 	idleConnTimeout := 90 * time.Second
 
 	return ClientConfig{
+		Headers:         headers,
```

Comment:
Why instantiate a variable `headers`?
```suggestion
		Headers:         make(map[string]configopaque.String),
```

Reply:
Yes it will be just waste of space and memory. P.S - corrected it :sweat_smile: